addons/website_sale_slides/controllers/slides.py

```diff
@@ -21,7 +21,7 @@ def get_course_products(self):
     def _prepare_additional_channel_values(self, values, **kwargs):
         values = super(WebsiteSaleSlides, self)._prepare_additional_channel_values(values, **kwargs)
         channel = values['channel']
-        if channel.enroll == 'payment' and channel.product_id:
+        if channel.enroll == 'payment' and channel.product_id.sudo().is_published:
```

Comment:
"sudo" call does not seem necessary.

I guess this is used to hide those information for the admin? (or someone that has access to the product record even though it's not published) ?

Reply:
Then maybe we could try the following to avoid introducing a sudo?

```py
if channel.enroll == 'payment':
    # re-browse the product to apply ACLs, notably on published status, to avoid access errors
    product = request.env['product.product'].browse(channel.product_id.id) if channel.product_id else request.env['product.product']
    if product.exists():
        # ... 
```

Can you give it a try and check that it fixes your use-case?

Idea is to avoid reading the product if you don't have access to it.
Because here your sudo will bypass more than just the published status, it will bypass everything.

Reply:
Seems a browse don't apply ACLs, so I used a search instead. What do you think ?

Reply:
That works too, but then we probably don't need to call "exists" :)